### PR TITLE
Fix: add nonce to give clear cache setting

### PR DESCRIPTION
**Pull request by jonwaldstein**
## Description

This adds CSRF protection to the GiveWP advanced setting "Clear Cache" by adding and verifying a nonce.

## Affects

The GiveWP advanced setting "Clear Cache" logic.

## Visuals

N/A

## Testing Instructions

Reach out for further instructions.

## Pre-review Checklist

-   [ ] Acceptance criteria satisfied and marked in related issue
-   [x] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit tests
-   [ ] Reviewed by the designer (if follows a design)
-   [ ] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204090406106770

### CODE REVIEW

**Comment by jonwaldstein** [comment]
@pauloiankoski @kjohnson this is ready for review.

**Review by kjohnson — APPROVED** [review]
Looks good to me.

**Review by pauloiankoski — APPROVED** [review]

File: includes/admin/settings/class-settings-advanced.php [hunk 1]
```diff
@@ -310,19 +310,30 @@ public function get_sections() {
...
+            echo !empty($field['wrapper_class']) ? 'class="' . $field['wrapper_class'] . '"' : ''; ?>>
+                <th scope="row" class="titledesc">
+                    <label
+                            for="<?php
+                            echo esc_attr($field['id']); ?>"><?php
+                        echo esc_html($field['name']); ?></label>
+                </th>
+                <td class="give-forminp">
+                    <button type="button" id="<?php
+                    echo esc_attr($field['id']); ?>"
```
Comment (ravinderk) [inline includes/admin/settings/class-settings-advanced.php:323]: @jonwaldstein Is this file formatted correctly?

Comment (jonwaldstein) [reply includes/admin/settings/class-settings-advanced.php]: resolved ab3d43c2

**Review by Benunc — APPROVED** [review]
This passes all manual tests, as well as automated tests.

**Pull request merged**